monty/events.py

```diff
@@ -0,0 +1,52 @@
+# from pydantic import dataclasses
+import dataclasses
+import enum
+import functools
+
+import disnake
+
+from monty.utils.code import prepare_input
+from monty.utils.markdown import remove_codeblocks
+from monty.utils.messages import extract_urls
+
+
+class MontyEvent(enum.Enum):
+    monty_message_processed = "monty_message_processed"
+
+
+@dataclasses.dataclass()
+class MessageContext:
+    """A context for a message event."""
+
+    content: str
+
+    # Lazily initialize attributes
+    @functools.cached_property
```

Comment:
If all these properties are cached, it might make sense to make the dataclass frozen too